FIAT/discontinuous_lagrange.py

```diff
@@ -5,7 +5,110 @@
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 
+import numpy as np
 from FIAT import finite_element, polynomial_set, dual_set, functional, P0
+from FIAT.finite_element import make_barycentric_lattice_coordinates, make_index_permutations
+
+
+def make_entity_permutations(dim, npoints):
+    if npoints <= 0:
+        return {o: [] for o in range(np.math.factorial(dim + 1))}
+    # DG nodes are numbered, in order of significance,
+    # - by g0: entity dim (vertices first, then edges, then ...)
+    # - by g1: entity ids (DoFs on entities of smaller ids first)
+    # - lexicographically as in CG
+    #
+    # Example:
+    # dim = 2, degree = 3 (npoints = degree + 1)
+    #
+    #     facet ids    CG node numbers  DG node numbers
+    #    +
+    #    | \           3                1
+    #    |   \  0      2 6              8 3
+    #  2 |     \       1 5 8            7 9 4
+    #    |       \     0 4 7 9          0 5 6 2
+    #    +--------+
+    #        1
```

Comment:
This is very confusing.

"DG degrees of freedom are numbered to geometrically coincide with CG degrees of freedom."

But note that the picture is wrong, because the reference cell is:

```
2
| \
|  \
1   0
|      \
|         \
0----2----1
```

Reply:
Fixed.